tpot/gp_deap.py

```diff
@@ -218,6 +233,10 @@ def eaMuPlusLambda(population, toolbox, mu, lambda_, cxpb, mutpb, ngen, pbar,
         if per_generation_function is not None:
             per_generation_function()
 
+        # Update generation statistic for all individuals
+        for ind in population:
+            ind.statistics['generation'] = gen
```

Comment:
This update procedure might be problematic for individuals that stick around from the previous generation. In our current GP algorithm, we don't have complete turnover between each iteration of the GP.

Reply:
What is this `ind.statistics['generation']` use for? I think some description about it is needed. If it is about which generation(s) has this individual, it should be a list instead.

Reply:
I believe the `generation` statistic is meant to track the generation that the individual first appeared.

Good point though, @weixuanfu. Since we're moving forward with this change, this PR should add some documentation (in the docstrings and the documentation) about these new statistics.

Reply:
Since the information in statistics only gets written to 'evaluted_individuals_' upon evaluation, and an individual is evaluated in the generation that it first occurs, the statistics reported in 'evaluated_individuals_' is always the information of the individual at the time of evaluation. In this light i did not build in any checks here, however, for debugging purposes it might be a good idea to indeed update 'generation' only when it concerns a new individual, and not update for individuals that traverse from the previous generation. 

I can implement this if you like, let me know.

Reply:
Because the behavior you described is a function of other parts of TPOT that are not directly related to the statistics tracking, which could change at any point and break this assumption, let's add an explicit check here to make sure that persistent individuals don't have their `generation` statistic updated. Please also add a comment above that check describing why we're doing that for future developers.

Reply:
Got it. How about incrementing the generation statistic along with the other statistics, at the moment the new individual is generated by mutation/cross-over?

Reply:
Yes, we could do that as long as the `generation` is available to that function.

Reply:
Yes, we could do that as long as the `generation` is available to that function.

Reply:
Ah i see, it's not Always generation = generation + 1.